aicsimageio/tests/test_aics_image.py

```diff
@@ -619,3 +617,278 @@ def test_roundtrip_save_all_scenes(
+            "Image:0",
+            None,
+            None,
+            dimensions.DEFAULT_DIMENSION_ORDER_WITH_SAMPLES,
+            ["Channel:0:0"],
+            (72, 1, 1, 268, 268, 4),
+        ),
+        # Check just dims to see default channel name creation
+        (
+            "example.gif",
+            "Image:0",
+            "ZYXC",
+            None,
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["Channel:0:0", "Channel:0:1", "Channel:0:2", "Channel:0:3"],
+            (1, 4, 72, 268, 268),
+        ),
+        # Check setting both as simple definitions
+        (
+            "example.gif",
+            "Image:0",
+            "ZYXC",
+            ["Red", "Green", "Blue", "Alpha"],
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["Red", "Green", "Blue", "Alpha"],
+            (1, 4, 72, 268, 268),
+        ),
+        # Check providing too many dims
+        pytest.param(
+            "example.gif",
+            "Image:0",
+            "ABCDEFG",
+            None,
+            None,
+            None,
+            None,
+            marks=pytest.mark.raises(exceptions=exceptions.ConflictingArgumentsError),
+        ),
+        # Check providing too many channels
+        pytest.param(
+            "example.gif",
+            "Image:0",
+            "ZYXC",
+            ["A", "B", "C"],
+            None,
+            None,
+            None,
+            marks=pytest.mark.raises(exceptions=exceptions.ConflictingArgumentsError),
+        ),
+        # Check providing channels but no channel dim
+        pytest.param(
+            "example.gif",
+            "Image:0",
+            None,
+            ["A", "B", "C"],
+            None,
+            None,
+            None,
+            marks=pytest.mark.raises(exceptions=exceptions.ConflictingArgumentsError),
+        ),
+        ######################################
+        # TiffReader
+        # First check to show nothing changes
+        (
+            "s_1_t_10_c_3_z_1.tiff",
+            "Image:0",
+            None,
+            None,
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["Channel:0:0", "Channel:0:1", "Channel:0:2"],
+            (10, 3, 1, 325, 475),
+        ),
+        # Check just dims to see default channel name creation
+        (
+            "s_1_t_10_c_3_z_1.tiff",
+            "Image:0",
+            "ZCYX",
+            None,
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["Channel:0:0", "Channel:0:1", "Channel:0:2"],
+            (1, 3, 10, 325, 475),
+        ),
+        # Check setting both as simple definitions
+        (
+            "s_1_t_10_c_3_z_1.tiff",
+            "Image:0",
+            "ZCYX",
+            ["A", "B", "C"],
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["A", "B", "C"],
+            (1, 3, 10, 325, 475),
+        ),
+        # Check setting channels as a list of lists definitions
+        (
+            "s_1_t_10_c_3_z_1.tiff",
```

Comment:
Just a thought - I wonder if these could be provided as fixtures that have a bunch of attributes attached to them. That way you don't have to remember `s_1_t_10_c_3_z_1.tiff` is `1x3x10x325x475` for every test and you can just use that attribute of the fixture object?

Reply:
I think in general I agree that would be useful but should be its own PR.

Reply:
Totally with you, that would be a fairly major overhaul!